nixarr/monitoring/default.nix

```diff
@@ -0,0 +1,263 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.nixarr;
+  
+  # Helper function to determine if a specific exporter should be enabled
+  shouldEnableExporter = service: 
+    cfg.${service}.enable && 
+    (cfg.monitoring.exporters.${service} or true);
+    
+  # Helper function to determine if a service is VPN-confined
+  isVpnConfined = service: cfg.${service}.enable && cfg.${service}.vpn.enable;
+    
+  # Helper function to create a script that extracts API key from config.xml
+  extractApiKeys = pkgs.writeShellApplication {
+    name = "extract-monitoring-api-keys";
+    runtimeInputs = with pkgs; [ dasel coreutils ];
+    text = ''
+      # Create state directory for API keys
+      STATE_DIR="/var/lib/exportarr"
```

Comment:
I assume this does not follow the other stateDir convention since this is not really stateful. Meaning that the information here is automatically generated deterministically from all the other statedirs, and as such doesn't need to be backed up in any way.

Reply:
Yes, that is correct.